test/features/manage_account/presentation/notification/notification_controller_test.dart

```diff
@@ -0,0 +1,161 @@
+import 'package:core/data/network/config/dynamic_url_interceptors.dart';
+import 'package:core/presentation/resources/image_paths.dart';
+import 'package:core/presentation/utils/app_toast.dart';
+import 'package:core/presentation/utils/responsive_utils.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get/get.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:tmail_ui_user/features/caching/caching_manager.dart';
+import 'package:tmail_ui_user/features/login/data/network/interceptors/authorization_interceptors.dart';
+import 'package:tmail_ui_user/features/login/domain/usecases/delete_authority_oidc_interactor.dart';
+import 'package:tmail_ui_user/features/login/domain/usecases/delete_credential_interactor.dart';
+import 'package:tmail_ui_user/features/manage_account/data/local/language_cache_manager.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/state/get_app_notification_setting_cache_state.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/state/toggle_app_notification_setting_cache_state.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/usecases/attempt_toggle_system_notification_setting_interactor.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/usecases/get_app_notification_setting_cache_interactor.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/usecases/log_out_oidc_interactor.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/usecases/toggle_app_notification_setting_cache_interactor.dart';
+import 'package:tmail_ui_user/features/manage_account/presentation/menu/settings/settings_controller.dart';
+import 'package:tmail_ui_user/features/manage_account/presentation/notification/notification_controller.dart';
+import 'package:tmail_ui_user/main/bindings/network/binding_tag.dart';
+import 'package:uuid/uuid.dart';
+
+import '../../../email/presentation/controller/single_email_controller_test.mocks.dart';
```

Comment:
We should use mock tests independently of test files. Avoid having to make many changes each time you revise a test case.

Reply:
Please give example which separating mock files will benefit

Reply:
Please delete some mocked objects in single_email_controller. Now you have to change some test files that used it.

Reply:
Mocks separated